contrib/check_inst_headers.sh

```diff
@@ -27,7 +27,7 @@ do
 	fi
 
 	# try to compile a test program (from stdin) which includes hfile
-	${CC} -I. -x c -c - -o /dev/null -DHAVE_CONFIG_H=1 <<EOF || exit $?
+	${CC} -I. -x c -c - -o /dev/null -DUCS_MAX_LOG_LEVEL=UCS_INFO <<EOF || exit $?
```

Comment:
removing DHAVE_CONFIG_H=1 essentially voids this entire test for all other headers!
i would set default value for UCS_MAX_LOG_LEVEL in log.h like this
```
#ifndef UCX_MAX_LOG_LEVEL
#  define UCS_MAX_LOG_LEVEL  UCS_LOG_LEVEL_TRACE_LAST
#endif
```

Reply:
the problem is not with UCS_MAX_LOG_LEVEL, config.h isn't available which together with DHAVE_CONFIG_H=1 leads to fail at  #include<config.h> in log.h. If there is no config.h why we need to define that we have it? Does it make sense to exclude log.h from this test?

Reply:
so we want to avoid installing files which include config.h
maybe split log.h to two files: log_def.h which will be installed and not include config.h, and log.h will just include config.h and after that log_def.h?